src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfClones = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...action.extraData,
+        };
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
+
+        for (const key of action.keysToRemove) {
+          if (currentState.hasOwnProperty(key)) {
+            delete currentState[key];
+          }
+        }
+        break;
+      case 'clear':
+        currentState = {};
+        break;
+      default:
+        throw new Error(`Unknown action type: ${action.type}`);
```

Comment:
Add empty line between cases
```suggestion
    switch (action.type) {
      case 'addProperties':
        currentState = {
          ...currentState, ...action.extraData,
        };
        break;
        
      case 'removeProperties':
        currentState = { ...currentState };

        for (const key of action.keysToRemove) {
          if (currentState.hasOwnProperty(key)) {
            delete currentState[key];
          }
        }
        break;
        
      case 'clear':
        currentState = {};
        break;
        
      default:
        throw new Error(`Unknown action type: ${action.type}`);
```